internal/services/nss/nss.go

```diff
@@ -151,6 +143,22 @@ func (s Service) GetShadowEntries(ctx context.Context, req *authd.Empty) (*authd
 	return &r, nil
 }
 
+// userPreCheck checks if the user exists in at least one broker.
+func (s Service) userPreCheck(ctx context.Context, username string) error {
+	// Check if the user exists in at least one broker.
+	for _, b := range s.brokerManager.AvailableBrokers() {
+		// The local broker is not a real broker, so we skip it.
+		if b.ID == brokers.LocalBrokerName {
+			continue
+		}
+		if err := b.UserPreCheck(ctx, username); err != nil {
+			continue
+		}
```

Comment:
Just a minor thing, and probably not worth it as I don't expect us having many brokers around, but maybe we could do these requests in parallel and the first one who replies wins?

Reply:
We don't care about which broker replies, only if one does, so I think it's not worth the complexity. Also, we would need to wait for all calls to complete to avoid having orphan routines so it might take more time than it should've in the end, no?